benchmarking/vllm_online_benchmark.py

```diff
@@ -0,0 +1,128 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+
+import os
+import subprocess
+import time
+import logging
+from datetime import datetime
+from typing import Dict
+from pathlib import Path
+
+from benchmarking.prompt_client_online_benchmark import get_test_combinations
+from utils.prompt_configs import EnvironmentConfig
+from utils.prompt_client import PromptClient
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+def run_benchmark(
+    params: Dict[str, int],
+    model: str,
+    port: int,
+    benchmark_script: str,
+    result_filename: Path,
+) -> None:
+    """Run a single benchmark with the given parameters."""
+    # fmt: off
+    cmd = [
+        "python", benchmark_script,
+        "--backend", "vllm",
+        "--model", model,
+        "--port", str(port),
+        # "--request-rate", "3",
+        "--dataset-name", "random",
+        "--num-prompts", str(params["num_prompts"]),
+        "--random-input-len", str(params["input_len"]),
+        "--random-output-len", str(params["output_len"]),
+        "--ignore-eos",  # Ignore EOS tokens to force max output length as set
+        "--save-result",
+        "--result-filename", str(result_filename)
+    ]
+    # fmt: on
+
+    logger.info(f"Running benchmark with parameters: {params}")
+    logger.info(f"Command: {' '.join(cmd)}")
+
+    try:
+        subprocess.run(cmd, check=True)
+        logger.info("Benchmark completed successfully")
+    except subprocess.CalledProcessError as e:
+        logger.error(f"Benchmark failed with error: {e}")
+    except Exception as e:
+        logger.error(f"Unexpected error during benchmark: {e}")
+
+    # Add a small delay between runs to ensure system stability
+    time.sleep(2)
+
+
+def main():
+    # Configuration
+    env_config = EnvironmentConfig()
+
+    # Create output directory
+    cache_dir = Path(os.environ.get("CACHE_ROOT", ""))
+    timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    result_dir = cache_dir / "vllm_online_benchmark_results" / f"results_{timestamp}"
+    result_dir.mkdir(parents=True, exist_ok=True)
+
+    prompt_client = PromptClient(env_config)
+    # note: there isnt a better way to pass an api key to the vllm benchmarking script
+    os.environ["OPENAI_API_KEY"] = prompt_client._get_authorization()
+
+    # Define benchmarking context length (isl, osl) pairs
+    context_lens = [
```

Comment:
Does this need to be coordinated with what is set in `prompt_client_online_benchmark.py`?

Reply:
No, only the vLLM benchmark script https://github.com/tenstorrent/vllm/blob/dev/benchmarks/benchmark_serving.py uses this authorization method.

Reply:
Sorry, I meant the `context_lens`

Reply:
They dont need to be but if for example we want to run the same context ISL/OSL pairs for max batch (using vllm benchmarking script) and single user with prompt_client_online_benchmark.py then they would need to be the same.